framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -896,31 +910,63 @@ protected String dotOutputFileName(UnderlyingAST ast) {
         return null;
     }
 
+    /**
+     * Returns the type of the left-hand side of an assignment without
+     * applying local variable defaults to type variables.
+     *
+     * The type variables that are types of local variables are defaulted to
```

Comment:
Type variables are [defaulted to top (or whatever the default for local variables is)](https://github.com/typetools/checker-framework/blob/master/framework/src/org/checkerframework/framework/util/defaults/QualifierDefaults.java#L461).  This special handling is for type argument inference.